core/trino-main/src/main/java/io/trino/execution/SetDefaultValueTask.java

```diff
@@ -0,0 +1,119 @@
+import io.trino.sql.PlannerContext;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.NodeRef;
+import io.trino.sql.tree.Parameter;
+import io.trino.sql.tree.SetDefaultValue;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.util.concurrent.Futures.immediateVoidFuture;
+import static io.trino.execution.ParameterExtractor.bindParameters;
+import static io.trino.metadata.MetadataUtil.createQualifiedObjectName;
+import static io.trino.spi.StandardErrorCode.COLUMN_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.spi.connector.ConnectorCapabilities.DEFAULT_COLUMN_VALUE;
+import static io.trino.sql.analyzer.ExpressionAnalyzer.analyzeDefaultColumnValue;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static java.util.Objects.requireNonNull;
+
+public class SetDefaultValueTask
+        implements DataDefinitionTask<SetDefaultValue>
+{
+    private final PlannerContext plannerContext;
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public SetDefaultValueTask(PlannerContext plannerContext, AccessControl accessControl)
+    {
+        this.plannerContext = requireNonNull(plannerContext, "plannerContext is null");
+        this.metadata = plannerContext.getMetadata();
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "SET DEFAULT";
+    }
+
+    @Override
+    public ListenableFuture<Void> execute(
+            SetDefaultValue statement,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        Session session = stateMachine.getSession();
+        Map<NodeRef<Parameter>, Expression> parameterLookup = bindParameters(statement, parameters);
+        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTableName());
+        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, tableName);
+
+        if (redirectionAwareTableHandle.tableHandle().isEmpty()) {
+            String exceptionMessage = "Table '%s' does not exist".formatted(tableName);
+            if (metadata.getMaterializedView(session, tableName).isPresent()) {
+                exceptionMessage += ", but a materialized view with that name exists.";
+            }
+            else if (metadata.isView(session, tableName)) {
+                exceptionMessage += ", but a view with that name exists.";
+            }
+            if (!statement.isTableExists()) {
+                throw semanticException(TABLE_NOT_FOUND, statement, "%s", exceptionMessage);
+            }
+            return immediateVoidFuture();
+        }
```

Comment:
Maybe as a follow-up - Do we need to abstract this out for all the DDL operations to be perfomed on a table ?